go/arrow/cdata/test/test_export_to_cgo.py

```diff
@@ -91,5 +93,22 @@ def test_record_batch(self):
             cgotest.importRecordBatch(self.ptr_schema, self.ptr_array)
 
 
+class TestGoToPython(BaseTestGoPython):
+
+    def test_get_schema(self):
+        with self.assert_pyarrow_memory_released():
+            cgotest.exportSchema(self.ptr_schema)
+
+            sc = pa.Schema._import_from_c(self.ptr_schema)
+            assert sc == self.make_schema()
+    
+    def test_get_batch(self):
+        with self.assert_pyarrow_memory_released():
```

Comment:
Is there also a way to check that all Go memory was released?

Reply:
We have an "allocator" object in the Go that I could expose a reference to for the tests which would then provide a way to check that the Go memory got freed. I'll see what i can do, but that's a really good point. It also ties into #11206 a bit where I set up a way to directly use and reference the `MemoryPool` in the C++ from Go

Reply:
added a `CurrentAlloc` function to the `CheckedAllocator` and have everything in the test use that in order to expose a function that allows confirming the Go memory has been released.